src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -121,41 +123,68 @@ public EntryEditor(BasePanel panel, ExternalFileTypes externalFileTypes) {
 
         this.tabs = createTabs();
 
-        this.setOnDragOver(event -> {
-            if (event.getDragboard().hasFiles()) {
-                event.acceptTransferModes(TransferMode.COPY, TransferMode.MOVE, TransferMode.LINK);
-            }
-            event.consume();
-        });
+        this.setOnMouseEntered(event -> {
+            // When registering using this.setOnDragDetected, the code is never called. Thus, using the workaround on "mouseEntered"
+            // onMouseEntered is called AFTER onDragDropped. We need it BEFORE. So we do this workaround with inDragDropMode
+            if (this.inDragDropMode) {
+                boolean dragDropModifierPressed = event.isAltDown() || event.isShiftDown() || event.isControlDown();
 
-        this.setOnDragDropped(event -> {
-            BibEntry entry = this.getEntry();
-            boolean success = false;
+                BibEntry entry = this.getEntry();
 
-            if (event.getDragboard().hasContent(DataFormat.FILES)) {
-                List<Path> files = event.getDragboard().getFiles().stream().map(File::toPath).collect(Collectors.toList());
                 FileDragDropPreferenceType dragDropPreferencesType = preferencesService.getEntryEditorFileLinkPreference();
 
-                if (dragDropPreferencesType == FileDragDropPreferenceType.MOVE) {
-                    LOGGER.debug("Mode MOVE");
-                    fileLinker.moveFilesToFileDirAndAddToEntry(entry, files);
-                    success = true;
+                // determine transfer mode -- JabRef's preferences are the default; modifier keys override the default
+                TransferMode determinedTransferMode;
+                if (dragDropModifierPressed) {
+                    LOGGER.debug("Modifier pressed");
+                    determinedTransferMode = transferMode;
+                } else {
+                    LOGGER.debug("No modifier pressed");
+                    switch (dragDropPreferencesType) {
+                        case LINK:
+                            determinedTransferMode = TransferMode.LINK;
```

Comment:
I think this is somehow duplicated code., an EnumMap would be the ideal solution here:
https://docs.oracle.com/javase/8/docs/api/java/util/EnumMap.html
So as keys you can use the "preferences" modes. and as values the real transfer mode and then you could simply return the the values for the keys.

It could be already initlaized in the preferences enum